google-cloud-tools-plugin/testSrc/com/google/cloud/tools/intellij/apis/CloudLibraryDependencyWriterTest.java

```diff
@@ -0,0 +1,159 @@
+      TestCloudLibraryClientMavenCoordinates.create(
+          CLOUD_LIB_GROUP_ID, CLOUD_LIB_ARTIFACT_ID, CLOUD_LIB_VERSION);
+  private static final TestCloudLibraryClient JAVA_CLIENT =
+      TestCloudLibraryClient.create(
+          "Client", "java", "API Ref", "alpha", "Source", "Lang Level", JAVA_CLIENT_MAVEN_COORDS);
+  private static final TestCloudLibrary LIBRARY =
+      TestCloudLibrary.create(
+          "Library",
+          "ID",
+          "service",
+          "Docs Link",
+          "Description",
+          "Icon Link 1",
+          ImmutableList.of(JAVA_CLIENT));
+
+  private MavenModuleBuilder moduleBuilder;
+
+  @Before
+  public void setUp() {
+    moduleBuilder = MavenTestUtils.getInstance().initMavenModuleBuilder(testFixture.getProject());
+  }
+
+  @Test
+  public void addLibraries_withNoBom_writesDependencyWithVersion() {
+    ApplicationManager.getApplication()
+        .invokeAndWait(
+            () -> {
+              MavenDomProjectModel model = getMavenDomProjectModel(null /*bomVersion*/);
+
+              List<MavenDomDependency> cloudLibraryDependencies =
+                  model.getDependencies().getDependencies();
+              assertThat(cloudLibraryDependencies.size()).isEqualTo(1);
+
+              MavenDomDependency dependency = cloudLibraryDependencies.get(0);
+              assertThat(dependency.getGroupId().getStringValue()).isEqualTo(CLOUD_LIB_GROUP_ID);
+              assertThat(dependency.getArtifactId().getStringValue())
+                  .isEqualTo(CLOUD_LIB_ARTIFACT_ID);
+              assertThat(dependency.getVersion().getStringValue()).isEqualTo(CLOUD_LIB_VERSION);
+            });
+  }
+
+  @Test
+  public void addLibraries_withBom_writesDependencyWithNoVersion() {
+    ApplicationManager.getApplication()
+        .invokeAndWait(
+            () -> {
+              String bomVersion = "1.0.0-beta";
+              MavenDomProjectModel model = getMavenDomProjectModel(bomVersion);
+
+              List<MavenDomDependency> cloudLibraryDependencies =
+                  model.getDependencies().getDependencies();
+              assertThat(cloudLibraryDependencies.size()).isEqualTo(1);
+
+              MavenDomDependency dependency = cloudLibraryDependencies.get(0);
+              assertThat(dependency.getGroupId().getStringValue()).isEqualTo(CLOUD_LIB_GROUP_ID);
+              assertThat(dependency.getArtifactId().getStringValue())
+                  .isEqualTo(CLOUD_LIB_ARTIFACT_ID);
+              assertThat(dependency.getVersion().getStringValue()).isNull();
+            });
+  }
+
+  @Test
+  public void addLibraries_withNoBom_doesNotWriteBom() {
+    ApplicationManager.getApplication()
+        .invokeAndWait(
+            () -> {
+              MavenDomProjectModel model = getMavenDomProjectModel(null /*bomVersion*/);
+
+              assertThat(model.getDependencyManagement().getDependencies().getDependencies())
```

Comment:
Mind bending call :) Could we probably expand this into a variable or two with a name so we can see what exactly is tested?

Reply:
lol. good idea

Reply:
done